core/cmd/client_test.go

```diff
@@ -329,3 +331,19 @@ func TestNewUserCache(t *testing.T) {
 	assert.DirExists(t, c.RootDir())
 
 }
+
+func TestNewApplication(t *testing.T) {
```

Comment:
Not super happy with this tbh... It's a pretty crummy test but was needed to get the test coverage % up. Any suggestions welcome!

Reply:
What does this cover that isn't already does in (dozens of) other tests? :thinking: 

Is our coverage calculation accurate? Or are we still missing out on cases where another package covers this?

Reply:
In this case it might be accurate? Factory.NewApplication is only called here in tests it looks like

Reply:
Oh this is actually part of the production subcommand to run the node, so it is left out by design b/c it does global config etc.

So it is in fact tested, but not via regular go tests that register coverage - just from `integration-tests/`.

But it also has grown too much. We should be able to get rid of the `EnsureChains` calls with https://smartcontract-it.atlassian.net/browse/BCF-2207

Reply:
Go tests which run nodes internally create a `TestApplication` here: https://github.com/smartcontractkit/chainlink/blob/587b7131c77dc222af1c40e966a77721d1c6b323/core/internal/cltest/cltest.go#L488

There is probably some duplication between the two paths that could be cleaned up to improve things all around, and increase internal coverage.